minpy/primitive.py

```diff
@@ -163,16 +202,46 @@ def def_grad_kw(self, func, key):
         :param key: Key name of the argument.
         :return: Self.
         """
-        self._grad_func_kw[key] = func
+        self._grad_func_kw[key] = GradFunc(f=func, multi_grad_indices=None)
         return self
 
     def def_grad_zero(self, argnum=0):
         """Define zero gradient
 
-        :param argnum: Index of the argument.
-        :return: Self.
+        Parameters
+        ----------
+        argnum
+            Index of the argument.
+
+        Returns
+        -------
+        Primitive
+            Self.
         """
-        self._grad_func[argnum] = lambda *args, **kwargs: lambda g: 0.0
+        self.def_grad(lambda *args, **kwargs: lambda g: 0.0, argnum)
+        return self
+
+    def def_multiple_grad(self, func, argnums):
+        """Define multiple gradients with one function.
+
+        Parameters
+        ----------
+        func
+            Gradient function.
+        argnums
+            Indices of the arguments.
+
+        Returns
+        -------
+        Primitive
+            Self.
+        """
+        assert isinstance(argnums, tuple), 'Indexes must be of type tuple.'
+        assert len(argnums) == len(set(argnums)), 'Duplicate entries.'
+
+        for argnum in argnums:
+            self._grad_func[argnum] = GradFunc(
+                f=func, multi_grad_indices=argnums)
```

Comment:
Hmm... I'm not sure about this. I thought we will only have a map in primitive like this:
```
arg_index_set + keyword_set => gradient function on them
```